experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -463,3 +500,108 @@ function Checkpointer.get_model_prog_state(sim::ClimaLandSimulation)
 end
 
 Interfacer.name(::ClimaLandSimulation) = "ClimaLandSimulation"
+
+## Extend functions for land-specific flux calculation
+"""
+    compute_surface_fluxes!(csf, sim::ClimaLandSimulation, atmos_sim, boundary_space, thermo_params, surface_scheme)
+
+This function computes surface fluxes between the integrated land model
+simulation and the atmosphere.
+This is intended to be used with the partitioned fluxes option.
+
+Update the input coupler surface fields `csf` in-place with the computed fluxes
+for this model. These are then summed using area-weighting across all surface
+models to get the total fluxes.
+
+Because the integrated land model is composed of multiple sub-components, the
+fluxes are computed for each sub-component and then combined to get the total for this model.
+The land model cache is updated with the computed fluxes for each sub-component.
+
+Any NaNs present in the fluxes are replaced by 0. Note that we are likely to
+get NaNs in areas where this model is evaluated but not physically present
+(e.g. land model in ocean grid cells).
```

Comment:
This is quite risky because it prevents possible problems from surfacing and being detected.

Reply:
Yeah I see your point. What we really want to do is filter out NaNs where the area fraction of a model is 0. I'll change it to do this tomorrow